build_resolvers/lib/src/resolver.dart

```diff
@@ -183,18 +184,39 @@ class AnalyzerResolvers implements Resolvers {
   /// Nullable, should not be accessed outside of [_ensureInitialized].
   Future<void> _initialized;
 
+  PackageConfig _packageConfig;
+
+  /// Lazily creates and manages a single [AnalysisDriver], that can be shared
+  /// across [BuildStep]s.
+  ///
+  /// If no [_analysisOptions] is provided, then an empty one is used.
+  ///
+  /// If no [sdkSummaryGenerator] is provided, a default one is used that only
+  /// works for typical `pub` packages.
+  ///
+  /// If no [_packageConfig] is provided, then one is created from the current
+  /// [Isolate.packageConfig].
+  ///
+  /// **NOTE**: The [_packageConfig] is used only for language versions, we
```

Comment:
Given our previous experience allowing these types to leak into our interfaces and then later migrating - should we consider some sort of `Map` here instead? It could also reduce the confusion leading to this note.

Reply:
talked offline, clarified the comment but left the type as its really only us who use this api and it allows for the most long term flexibility if new required fields are added.